src/common/types/TableTypes.types.ts

```diff
@@ -0,0 +1,64 @@
+import * as React from 'react';
+import { ChangeEvent, MouseEvent } from 'react';
+import { SortingDirection } from '../../common/utils/SortingDirection';
+
+export interface RoomData {
+  id: number;
+  number: number;
+  floor: number;
+  type: string;
+  keyOwner: string;
+  openingHours: string;
+  capacity: number;
+}
+
+export interface UserData {
+  id: number;
+  name: string;
+  surname: string;
+  email: string;
+  room: number;
+  userType: string;
+}
+
+export interface HeadCell<T extends { id: number }> {
+  disablePadding: boolean;
+  id: keyof T;
+  label: string;
+}
+
+export interface CustomTableToolbarProps<T extends { id: number }> {
+  allSelected: readonly number[];
+  allRows: readonly T[];
+}
+export interface CustomTableHeadProps<T extends { id: number }> {
+  numSelected: number;
+  onRequestSort: (event: MouseEvent<unknown>, property: keyof T) => void;
+  onSelectAllClick: (event: ChangeEvent<HTMLInputElement>) => void;
+  order: SortingDirection;
+  orderBy: string;
+  rowCount: number;
+  // onSortCallback: (headCellName: string, direction: SortingDirection) => void; // will be used later, when we synchronize lists with backend
+}
```

Comment:
this constraint is too strict. We should use `Record<PropertyKey, unknown>`

Reply:
Replaced the constraint in [92390f3](https://github.com/dorm-social-room-manager/dsrm-frontend/pull/38/commits/92390f35bc955ce6069d373c264463c3e29f685a)